evo_researcher/functions/utils.py

```diff
@@ -0,0 +1,9 @@
+import tiktoken
+
+
+def trim_to_n_tokens(content: str, n: int, model: str, trim_batch_size: int = 128) -> str:
+    encoder = tiktoken.encoding_for_model(model)
+    encoded = encoder.encode(content)
+    while len(encoded) > n:
```

Comment:
Why not 
```
encoded = encoder.encode(content)
encoded = encoded[:n]
return encoder.decode(encoded)
```
?

Reply:
Probably some kind of brain fog 😄 Updated of course